File: leetcode_problems/python/lc1415_kth_lexigraphical_string_of_all_happy_strings.py
class Solution:
    def getHappyString(self, n: int, k: int) -> str:
        def backtrack() -> bool:
            nonlocal count
            if len(stack) >= n:
                count += 1
                return count == k

            for char in "abc":
                if len(stack) > 0 and stack[-1] == char:
                    continue

                stack.append(char)
                if backtrack():
                    return True
                stack.pop()

            return False

        count = 0
        stack = []
        backtrack()
        return "".join(stack)


if __name__ == "__main__":
    n = 3
    k = 9
    print(Solution().getHappyString(n, k))

File: leetcode_problems/typescript/src/lc1415_kth_lexigraphical_string_of_all_happy_strings.ts
function getHappyString(n: number, k: number): string {
  function backtrack(): boolean {
    if (stack.length >= n) {
      count++;
      return count == k;
    }

    for (let char of "abc") {
      if (stack.length > 0 && stack[stack.length - 1] == char) {
        continue;
      }

      stack.push(char);
      if (backtrack()) {
        return true;
      }
      stack.pop();
    }
    return false;
  }

  let count = 0;
  let stack: string[] = [];
  backtrack();
  return stack.join("");
}
